homeassistant/components/madvr/sensor.py

```diff
@@ -277,4 +277,15 @@ def __init__(
     @property
     def native_value(self) -> float | str | None:
         """Return the state of the sensor."""
-        return self.entity_description.value_fn(self.coordinator)
+        val = self.entity_description.value_fn(self.coordinator)
+        # check if sensor is enum
+        if self.entity_description.device_class == SensorDeviceClass.ENUM:
+            if (
+                self.entity_description.options
+                and val in self.entity_description.options
+            ):
+                return val
+            # return None for values that are not in the options
+            return None
+
+        return val
```

Comment:
This only applies for one sensor? Honestly, just add it to the value_fn

Reply:
theres at least two

Reply:
I think doing it this way is cleaner regardless there are two placeholder values that I know of but 11 enums and this issue may come up unexpectedly in the future anyhow.

Reply:
yeah thought about it more, this approach is cleaner and more explicit